File: array/medium/golang_solutions/minimum_size_subarray_sum_209.go
package golang_solutions

func MinSubArrayLen(target int, nums []int) int {
	start, end, currentSum, minLength := 0, 0, 0, 1<<31

	for end < len(nums) {
		currentSum += nums[end]
		end++
		for start < end && currentSum >= target {
			difference := end - start
			if minLength > difference {
				minLength = difference
			}
			currentSum -= nums[start]
			start++
		}
	}
	if minLength == 1<<31 {
		return 0
	}
	return minLength
}

File: leetcode/array/medium/java_solutions/MinimumSizeSubarraySum_209.java
package leetcode.array.medium.java_solutions;

public class MinimumSizeSubarraySum_209 {

    public int minSubArrayLen(int target, int[] nums) {
        int minLength = 1 << 30;
        int left = 0, right = 0, tempSum = 0;

        while (right < nums.length) {
            tempSum += nums[right];
            right++;
            while (left < right && tempSum >= target) {
                if (minLength > right - left) {
                    minLength = right - left;
                }
                tempSum -= nums[left++];
            }
        }
        return minLength == 1 << 30 ? 0 : minLength;
    }
}
